modules/Rock.cmake

```diff
@@ -1350,6 +1368,39 @@ function(rock_gtest TARGET_NAME)
 
     rock_setup_gtest_test(${TARGET_NAME} ${GMOCK_DIR} ${GTEST_DIR})
     rock_add_test(${TARGET_NAME} "${__rock_test_parameters}")
+    if (ROCK_TEST_CXX_GCOVR_GENERATION_ENABLED)
+        rock_coverage_report_generation(${PROJECT_NAME} ${TEST_TARGET_NAME})
```

Comment:
As you rightly noticed (from your usage of PROJECT_NAME), the coverage reports are per-project, not per-target. You should follow that and make the default behaviour:
- save the list of all test targets somewhere global
- call `rock_cxx_coverage_report` in rock_standard_layout after the test folder (https://github.com//rock-core/base-cmake/blob/6415b607b8368966c7dccb60ae706266f2ca29df/modules/Rock.cmake#L295), automatically adding all tests as dependencies.